src/Graph.ts

```diff
@@ -149,7 +154,13 @@ export default class Graph {
 
 		this.varOrConst = options.preferConst ? 'const' : 'var';
 		this.legacy = options.legacy;
+
 		this.acornOptions = options.acorn || {};
+		this.acornParse = ensureArray(options.acornInjectPlugins).reduce(
+			(acc, plugin) => plugin(acc),
+			acorn
+		).parse;
```

Comment:
Ideally we could add the dynamic import plugin via something like `[...(this.dynamicImport ? [injectDynamicImportPlugin] : []) , ...ensureArray(options.acornInjectPlugins)].reduce(...`, altering `wrapDynamicImport` to retun the acorn instance to match the pattern.

Not having the `dynamicImport` check on the plugin is my oversight though, no pressure to add that here you've done a lot already.

Reply:
Yeah, I was thinking yesterday that it would be nice to use the same system to inject both internal and user-defined plugins. With your recent changes, it should be easy to do. Let me try.

Reply:
Done in https://github.com/rollup/rollup/pull/1857/commits/d58553b589002fa0816bc3800007c1249b874ece. I wasn't sure how to declare the return type of `injectDynamicImportPlugin`, but leaving it undeclared worked too.